src/tracing/esp32_diagnostic_trace/Counter.cpp

```diff
@@ -0,0 +1,60 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <esp_log.h>
+#include <tracing/esp32_diagnostic_trace/Counter.h>
+
+namespace chip {
+namespace Tracing {
+namespace Diagnostics {
+
+std::map<const char *, uint32_t> ESPDiagnosticCounter::mCounterList;
```

Comment:
these compares will not be `strcmp` like compares but rather memory pointer compares. Is this expected/intended?